hollow/src/main/java/com/netflix/hollow/api/codegen/HollowConsumerJavaFileGenerator.java

```diff
@@ -63,9 +85,42 @@ protected void appendPackageAndCommonImports(StringBuilder builder) {
             }
 
             if (config.isUsePackageGrouping()) {
-                builder.append("import ").append(packageName).append(".*;\n");
-                builder.append("import ").append(packageName).append(".core.*;\n");
-                if (useCollectionsImport) builder.append("import ").append(packageName).append(".collections.*;\n\n");
+                if (apiClassname != null) {
+                    appendImportFromBasePackage(builder, apiClassname);
+                }
+                Set<String> imports = new HashSet<>();
+                for (HollowSchema schema : schemasToImport) {
+                    switch (schema.getSchemaType()) {
+                        case OBJECT:
+                            addToSetIfNotPrimitive(imports, schema.getName());
+                            break;
+                        case SET:
+                            addToSetIfNotPrimitive(imports,
+                                    ((HollowSetSchema) schema).getElementType());
+                            break;
+                        case LIST:
+                            addToSetIfNotPrimitive(imports,
+                                    ((HollowListSchema) schema).getElementType());
+                            break;
+                        case MAP:
+                            HollowMapSchema mapSchema = (HollowMapSchema) schema;
+                            addToSetIfNotPrimitive(imports, mapSchema.getKeyType(),
+                                    mapSchema.getValueType());
+                            break;
+                        default:
+                            throw new IllegalArgumentException(
+                                    "Unexpected HollowSchema to import: " + schema);
+
+                    }
+                    for (String anImport : imports) {
```

Comment:
Not sure if I am reading this correctly, but should this be outside of the for loop?

Reply:
Ahh you're right, this is incorrectly inside the for loop. Probably resulted in extra imports, which I didn't notice :/